python/python/lance/dataset.py

```diff
@@ -3043,6 +3058,33 @@ class Project(BaseOperation):
         schema: LanceSchema
 
 
+class ColumnOrdering:
+    """
+    This class is used to define the column ordering rules for the `sort` operator.
+    It allows users to specify the sorting order (ascending or descending)
+    and the position of null values (first or last).
+    """
```

Comment:
Two pieces of feedback:

1. Instead of all these static methods, it would be more pythonic to just use keyword arguments for `ascending` and `nulls_first`, like:  `ColumnOrdering("int_col")`, `ColumnOrdering("str_col", ascending=False)`, `ColumnOrdering("data", nulls_first=True)`.
2. For classes that are just configuration, I'd rather just make them `dataclass` in Python and then use the `FromPyObject` / `ToPyObject` marhsalling in Rust. You can see how we've done this for `RewrittenIndex`, for example.

https://github.com/lancedb/lance/blob/18f20c3cf6c4bc167d8f76c6718dbfc25ebccd46/python/python/lance/dataset.py#L2942-L2949

https://github.com/lancedb/lance/blob/18f20c3cf6c4bc167d8f76c6718dbfc25ebccd46/python/src/transaction.rs#L320-L348

```suggestion
@dataclass
class ColumnOrdering:
    """
    This class is used to define the column ordering rules for the `sort` operator.
    It allows users to specify the sorting order (ascending or descending)
    and the position of null values (first or last).
    """
    column_name: str
    ascending: bool = True
    nulls_first: bool = False
```